HACKING.md

```diff
@@ -224,10 +224,51 @@ OpenShift is checked into this repository.  To install `godep` locally run:
 
 If you are not updating packages you should not need godep installed.
 
-## Cherry-picking an upstream commit into Origin
+## Cherry-picking an upstream commit into Origin: Why, how, and when.
+
+Upstream origin carries patches inside of vendor/ on top of each rebase.  Thus, Origin consumes kubernetes in two ways.
+
+1. Periodic rebases against a kubernetes commit.  Eventually, any patch you have in upstream kubernetes will land in openshift via this mechanism.
+
+2. Cherry picked patches for important features.  This is done when an important feature/fix is in a version of kubernetes that we have not yet rebased against.
+
+### Manually
+
+You can manually try to cherry pick a commit, by using git apply. This can easily be done in a couple of steps.
+
+- wget the patch, i.e. `wget -O /tmp/mypatch https://github.com/kubernetes/kubernetes/pull/34624.patch`
+- PATCH=/tmp/mypatch git apply --directory vendor/k8s.io/kubernetes $PATCH
```

Comment:
Not sure about others but I always let the patch go to <pr number>.patch in my current directory and apply there.  Using the output file and then a subsequent env var seems workable but maybe not what I'd consider normal.  I don't feel strongly - this definitely works, just caught my eye.